### DISCUSSION TITLE
Should cue.mod/gen files be committed when using tools such as Bazel?

### DISCUSSION THREAD

**Discussion by elyzion** [cuelang/cue#775] [General]
My team uses Bazel for build management. Generally speaking generated files are not commited, and are generated as needed during compilation and/or execution by the Bazel tooling. 

Currently there is some discussion in the team as to whether cue.mod/gen files should be committed, or if they should be treated the same as generated protobuf sources. 

My feeling is that for Cue these files should be committed, even if this is not the "pure" Bazel way. 

Does anyone have any experiences/opinions on this?

**Comment by mpvl** [comment]
I would tend to agree with you, that these should be committed.

The idea of `cue.mod` directory is to make a CUE evaluation hermetic, or devoid of outside influences. Generating such files may change as versions of the underlying tooling will change. One could assume that the generated results are always the same, but that is a rather big assumption.

For an analogy, I would attribute the same kind of status to `cue.mod/gen` files as files generated by `go generate`. The latter are typically expected to be checked in along with the other Go files. At least for the Bazel setups I have seen files generated by `go generated` are checked in.

**Comment by uhthomas** [comment]
FWIW this is what I'm doing for my Kubernetes cluster(s) https://github.com/uhthomas/automata/commit/3daf21253680c5a3a8c4e10be816bef704e7ae70.

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/775.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
